### PR TITLE
Add BondiSachs and deadlock analysis to SingleBH

### LINKED ISSUE #5672: Add CCE output to SingleBH executable
SingleBH exec needs waveforms for ringdown

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

BondiSachs is going to be needed for waveforms. Deadlock analysis is just a good thing to have and it makes it in line with the BBH executable.

Fixes #5672.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp [hunk 1]
```diff
@@ -243,13 +253,40 @@ struct EvolutionMetavars : public GeneralizedHarmonicTemplateBase<3> {
...
+  static void run_deadlock_analysis_simple_actions(
+      Parallel::GlobalCache<EvolutionMetavars>& cache,
+      const std::vector<std::string>& deadlocked_components) {
+    const auto& functions_of_time =
+        Parallel::get<::domain::Tags::FunctionsOfTime>(cache);
+
+    const std::string time_bounds =
+        ::domain::FunctionsOfTime::ouput_time_bounds(functions_of_time);
+
+    Parallel::printf("%s\n", time_bounds);
```
Comment (nilsdeppe) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp:268]: Maybe have it say "Time bounds for functions of time are: "?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, one suggestion. If you agree, squash immediately

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp (hunk 1 above)
Comment (knelli2) [reply src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp:268]: `output_time_bounds()` actually already has that in the string it returns.

Comment (nilsdeppe) [reply src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp:268]: Ah okay, sounds good! I'll merge as-is then :)

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**